### DISCUSSION TITLE
The specified path, file name, or both are too long.

### DISCUSSION THREAD

**Discussion by kkhor3** [wixtoolset/issues#7447] [Questions]
WIX Toolset version 3.11

Frequently hit into this error message in GitHub Action. Error message: 'The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.'

Details why it is hitting the limit. https://github.com/actions/runner/issues/1676 
We have enable LongPathsEnable (window registry) and core.longpaths (git config) as instructed from another discussion, which is working great in MSBuild but not for Wix.

My problem fixed by shortening the namespace, however I'm curious if anyone is seeking in enabling Wix to handle longer file paths and file names? 

Because of the nature of my company's environment, we have extensive namespaces or a deep folder hierarchy, which is difficult to alter across numerous repositories. I would greatly appreciate it if someone could help us in this matter.

**Comment by robmen** [comment]
The last investigation into supporting paths over MAX_PATH showed this wasn't as simple as it should be because some MSI APIs do not support long paths. This would be an honorable work item for someone to take, but I expect it's non-trivial.

**Reply by sumedhbhogle** [reply]
Hello @robmen 
Is this still a limitation. Installer created with WiX 4.0 fails and it seems due to target path for the component exceeds MAX_PATH limit. I could not see anywhere in MSI APIs documentation to conclude that this is still a limitation. 

Regards,
Sumedh

**Reply by robmen** [reply]
Are you asking me if MAX_PATH is still a limitation for the Windows Installer or are you asking why the documentation doesn't indicate that the Windows Installer doesn't support MAX_PATH?

**Reply by sumedhbhogle** [reply]
Sorry for confusion.

I am asking if MAX_PATH is still a limitation for the Windows Installer.

Regards,
Sumedh

**Reply by robmen** [reply]
That sounds like a question that would be better directed to Microsoft, which owns the Windows Installer (MSI). As I noted above, my experience is the same as yours.

Why did you think I would have a better answer?

**Reply by sumedhbhogle** [reply]
Understood!

Thanks @robmen 

Regards,
Sumedh

**Reply by robmen** [reply]
I'm glad you understood, but @sumedhbhogle, why did you think I would provide a better answer to your question than I already had? I'm trying to understand this interaction.

**Comment by bevanweiss** [comment]
References to Microsoft documentation on this.
https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

So this initial limitation would be an issue for the deployed application, as well as the actual installation process.  Since the Microsoft Windows Installer subsystem would need to opt into this long path handling... which from some testing I did around the RemoveFile handling was not the case.

There's a whole bunch of MSI specific methods that would also need to support the long path handling.
https://learn.microsoft.com/en-us/windows/win32/api/msi/

So there's lots that Microsoft would need to do to support long paths from the Windows Installer (and I'd imagine it would then need to be another version, so a Windows Installer 6.0 or such...).

Even if the Windows Installer (msi) methods did enable long path support (or there was a way to trick them).
There are likely quite a few WiX custom actions that would not just work out of the box, and would need some additional handling.
It would be quite a lot of work to set it all up and put in place the additional test cases for it all.  I doubt there is much excitement for someone to do this unless they were going to be paid for it...

Here's where you can see the various Windows Installer versions.
https://learn.microsoft.com/en-us/windows/win32/msi/what-s-new-in-windows-installer
If you find an explicit mention of long path support, please let me know..

**Comment by kartlee** [comment]
Hello,

We use WiX v3.7 to build MSIs, and were testing how MSI behaves when providing an installation path longer than 260 characters. We encountered the following error:

```
MSI (s) (18:48) [14:46:44:556]: Executing op: FileCopy(SourceName=xmyj3kcb.dll|GLWidgets.dll,SourceCabKey=F.4_bin_Windows_x64_GLWidgets.dll8cea4a20962b864ed03d65025137ed52,DestName=GLWidgets.dll,Attributes=512,FileSize=48640,PerTick=65536,,VerifyMedia=1,,,,,CheckCRC=0,Version=6.5.5.0,Language=1033,InstallMode=58982400,,,,,,,)
MSI (s) (18:48) [14:46:44:556]: File: E:\temp\longpath\test1\test2\test3\test4\test5\test6\test7\test8\test9\test10\test11\test12\test13\test14\test15\test16\test17\test18\test19\test20\test21\test22\test23\test24\test25\test26\test27\test28\test29\test30\test31\test32\test33\test34\test35\test36\test37\product-v8.4\bin\Windows-x64\GLWidgets.dll; To be installed; Won't patch; No existing file
MSI (s) (18:48) [14:46:44:556]: Note: 1: 1320 2: E:\temp\longpath\test1\test2\test3\test4\test5\test6\test7\test8\test9\test10\test11\test12\test13\test14\test15\test16\test17\test18\test19\test20\test21\test22\test23\test24\test25\test26\test27\test28\test29\test30\test31\test32\test33\test34\test35\test36\test37
MSI (s) (18:48) [14:46:44:556]: Note: 1: 2205 2: 3: Error
MSI (s) (18:48) [14:46:44:556]: Note: 1: 2228 2: 3: Error 4: SELECT `Message` FROM `Error` WHERE `Error` = 1320
MSI (c) (1C:20) [14:46:44:556]: Font created. Charset: Req=0, Ret=0, Font: Req=MS Shell Dlg, Ret=MS Shell Dlg

Error 1320. The specified path is too long: E:\temp\longpath\test1\test2\test3\test4\test5\test6\test7\test8\test9\test10\test11\test12\test13\test14\test15\test16\test17\test18\test19\test20\test21\test22\test23\test24\test25\test26\test27\test28\test29\test30\test31\test32\test33\test34\test35\test36\test37
```

Based on the above error, we assume this is something internal to MSI APIs that's still not fixed. Is this understanding correct? We tried providing the extended path format, and also have the longPathEnabled in the registry. Both failed to recognize the installation path.

We also assume that moving to v4 version of WiX is not going to solve this installation issue. Is this understanding correct?

-Karthik

**Reply by robmen** [reply]
> ...we assume this is something internal to MSI APIs that's still not fixed. 

As far as I am aware, Microsoft has not updated the Windows Installer to support long paths. You'd need to follow up with Microsoft to validate that.
